File: src/pocketcloud/scheduler/Task.php
<?php

namespace pocketcloud\scheduler;

abstract class Task {

    private ?TaskHandler $taskHandler = null;

    abstract public function onRun(): void;

    public function onCancel(): void {}

    public function setTaskHandler(TaskHandler $taskHandler): void {
        $this->taskHandler = $taskHandler;
    }

    public function getTaskHandler(): TaskHandler {
        return $this->taskHandler;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/scheduler/Task.java
package de.pocketcloud.cloud.scheduler;

import lombok.Getter;
import lombok.Setter;

public abstract class Task {

    @Setter
    @Getter
    private TaskHandler handler = null;

    abstract public void onRun(long currentTick);

    public void onCancel(long currentTick) {}
}